pkg/services/ngalert/api/api_ruler.go

```diff
@@ -263,6 +263,46 @@ func (srv RulerSrv) RouteGetRulesConfig(c *contextmodel.ReqContext) response.Res
 	return response.JSON(http.StatusOK, result)
 }
 
+// RouteGetRuleByUID returns the alert rule with the given UID
+func (srv RulerSrv) RouteGetRuleByUID(c *contextmodel.ReqContext, ruleUID string) response.Response {
+	ctx := c.Req.Context()
+	orgID := c.SignedInUser.GetOrgID()
+
+	rule, err := srv.getAuthorizedRuleByUid(ctx, c, ruleUID)
+	if err != nil {
+		if errors.Is(err, ngmodels.ErrAlertRuleNotFound) {
+			return response.Empty(http.StatusNotFound)
+		}
+		return response.ErrOrFallback(http.StatusInternalServerError, "failed to get rule by UID", err)
+	}
+
+	provenance, err := srv.provenanceStore.GetProvenance(ctx, &rule, orgID)
+	if err != nil {
+		return response.ErrOrFallback(http.StatusInternalServerError, "failed to get rule provenance", err)
+	}
+
+	result := &apimodels.GettableGrafanaRule{
+		ID:                   rule.ID,
+		OrgID:                rule.OrgID,
+		Title:                rule.Title,
+		Condition:            rule.Condition,
+		Data:                 ApiAlertQueriesFromAlertQueries(rule.Data),
+		Updated:              rule.Updated,
+		IntervalSeconds:      rule.IntervalSeconds,
+		Version:              rule.Version,
+		UID:                  rule.UID,
+		NamespaceUID:         rule.NamespaceUID,
+		RuleGroup:            rule.RuleGroup,
+		NoDataState:          apimodels.NoDataState(rule.NoDataState),
+		ExecErrState:         apimodels.ExecutionErrorState(rule.ExecErrState),
+		Provenance:           apimodels.Provenance(provenance),
+		IsPaused:             rule.IsPaused,
+		NotificationSettings: AlertRuleNotificationSettingsFromNotificationSettings(rule.NotificationSettings),
+	}
```

Comment:
I've not spent a lot of time in this code, but it looks like the other APIs here return a `GettableExtendedRuleNode`, which has some extra fields (`For`, `Annotations`, `Labels`).

I suspect we would want to do the same here for consistency. Could we re-use the `toGettableExtendedRuleNode` function?
```suggestion
	result := toGettableExtendedRuleNode(rule, provenance)
```